corehq/apps/case_search/xpath_functions/comparison.py

```diff
@@ -23,25 +23,34 @@ def property_comparison_query(context, case_property_name_raw, op, value_raw, no
 
     case_property_name = serialize(case_property_name_raw)
     value = unwrap_value(value_raw, context)
-    # adjust the user's input date based on project timezones
-    is_user_input = False
-    try:
-        # this might be inconsistent in daylight savings situations
-        value = adjust_input_date_by_timezone(value_to_date(node, value),
-                                              get_timezone_for_domain(context.domain), op)
-        is_user_input = True
-    except (XPathFunctionException, AssertionError):
-        # AssertionError is caused by tests that use domains without a valid timezone (in get_timezeone_for_domain)
-        pass
+    if case_property_name in SPECIAL_CASE_PROPERTIES:
```

Comment:
What would you think of further restricting this logic to only datetime properties?  I think right now it'll try to convert expressions like `name = 'Ethan'` to a datetime filter, which will fail at some point.  I think the `try/except` will still make it work correctly, but the logic is muddled.  `value_to_date` raises `XPathFunctionException` to provide feedback to the user that they were doing something invalid.  This instead uses that same exception to determine what kind of filtering to do.

For example, if the user does a query like `date_opened > '2023-02-31'`, that is an error because February doesn't have 31 days.  The correct response would be to inform the user that it's not a valid date.  Instead, this would fall back to a string query.  Also, it attempts to interpret integers as days since the unix epoch, which seems like it could easily get false positives
https://github.com/dimagi/commcare-hq/blob/a0cbc781ea8b00f6d054e4e12873da3d120829bf/corehq/apps/case_search/xpath_functions/value_functions.py#L33-L34

I can think of two easy ways to achieve this - make a list of the three SPECIAL_CASE_PROPERTIES that are datetimes, or better yet, add a datatype arg to `SpecialCaseProperty` and filter on that.